src/main/c/NativeLongFunc.cpp

```diff
@@ -0,0 +1,142 @@
+//
+// src/main/c/NativeFunc.cpp - native funcs for Scala NativeLongFunc
+//
+// Richard Cownie, Hail Team, 2018-04-18
+//
+#include "hail/NativePtr.h"
+#include "hail/CommonDefs.h"
+#include <assert.h>
+#include <jni.h>
+
+NAMESPACE_BEGIN(hail)
+
+NAMESPACE_BEGIN_ANON
+
+typedef NativeFuncObj<long> NativeFunc;
```

Comment:
We should prefer using over typedef.  From the style guide:

 > In new code, using is preferable to typedef, because it provides a more consistent syntax with the rest of C++ and works with templates.